src/platform/packages/shared/kbn-telemetry/src/init_telemetry.ts

```diff
@@ -27,23 +35,47 @@ export const initTelemetry = (
   const apmConfigLoader = loadConfiguration(argv, rootDir, isDistributable);
 
   const apmConfig = apmConfigLoader.getConfig(serviceName);
-
   const telemetryConfig = apmConfigLoader.getTelemetryConfig();
 
-  // explicitly check for enabled == false, as the default in the schema
-  // is true, but it's not parsed through @kbn/config-schema, so the
-  // default value is not returned
-  const telemetryEnabled = telemetryConfig?.enabled !== false;
+  // attributes.resource.*
+  const resource = resources.resourceFromAttributes({
+    [ATTR_SERVICE_NAME]: apmConfig.serviceName,
+    [ATTR_SERVICE_VERSION]: apmConfig.serviceVersion,
+    [ATTR_SERVICE_INSTANCE_ID]: apmConfig.serviceNodeName,
+    [ATTR_SERVICE_NAMESPACE]: apmConfig.environment,
+    ...(apmConfig.globalLabels as Record<string, unknown>),
+  });
 
-  // tracing is enabled only when telemetry is enabled and tracing is not disabled
-  const tracingEnabled = telemetryEnabled && telemetryConfig?.tracing?.enabled;
+  if (telemetryConfig.enabled) {
+    const desiredInstrumentations = new Set<keyof InstrumentaionsMap>();
 
-  if (!tracingEnabled) {
-    return async () => {};
-  }
+    if (telemetryConfig.tracing.enabled) {
+      initTracing({ resource, tracingConfig: telemetryConfig.tracing });
+    }
 
-  return initTracing({
-    tracingConfig: telemetryConfig.tracing,
-    apmConfig,
-  });
+    if (telemetryConfig.metrics.enabled) {
+      initMetrics({ resource, metricsConfig: telemetryConfig.metrics });
+
+      // Provides metrics about the Event Loop, GC Collector, and Heap stats.
+      desiredInstrumentations.add('@opentelemetry/instrumentation-runtime-node');
+      // HTTP Server and Client durations
+      desiredInstrumentations.add('@opentelemetry/instrumentation-http');
+      // Undici client's request duration
+      desiredInstrumentations.add('@opentelemetry/instrumentation-undici');
```

Comment:
We probably want to disable all of the tracing stuff for both http and undici. Is this possible?

Reply:
https://github.com/open-telemetry/opentelemetry-js/blob/e1ec4026edae53a2dea3a9a604d6d21bb5e8d99f/experimental/packages/opentelemetry-instrumentation-http/src/types.ts#L66-L68 ?

Reply:
Wouldn't that mean that they don't track the metrics of those requests?